### PR TITLE
[2019.2] Fixes to run_job in scheduler

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Updating the max_running function to assume run=True if the value isnot available, this fixes the run_job function.  Adding a test for run_job.

### What issues does this PR fix or reference?
#52617

### Tests written?
Yes

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/utils/schedule.py [hunk 1]
```diff
@@ -210,7 +210,7 @@ def _check_max_running(self, func, data, opts, now):
         # dict we treat it like it was there and is True
 
         # Check if we're able to run
-        if not data['run']:
+        if not data.get('run', True):
```
Comment (waynew) [inline salt/utils/schedule.py]: Why is it that we want to default to `True` here?

Comment (garethgreenaway) [reply salt/utils/schedule.py]: Updated the code, reverting this one back and setting the value in the run function.

**Comment by waynew** [comment]
Would it also make sense to change that check to `if not data.get('run')`? I haven't really spent enough time grokking this code to know if that's the "right" idea or not, though. 🤷‍♂️

**Comment by garethgreenaway** [comment]
@waynew Probably a good idea, in all cases it should be there but that would be a good "better safe than sorry" check.  Will update the PR.

**Review by waynew — APPROVED** [review]

**Comment by waynew** [comment]
Looks like there was a failure but like... :suspect: 2 seconds off? I'm not really sure how to handle that. Rerun the test? Update that test suite to better handle time?

**Comment by garethgreenaway** [comment]
@waynew Looks like 2 minutes off.  I've seen this particular error before. Taking a look.

File: tests/integration/scheduler/test_eval.py [hunk 2]
```diff
@@ -753,15 +769,19 @@ def test_eval_minutes(self):
         ret = self.schedule.job_status(job_name)
         self.assertEqual(ret['_last_run'], run_time)
 
-        time.sleep(2)
+        # Give the job a chance to finish
+        while self.schedule.job_running(ret):
```
Comment (dwoz) [inline tests/integration/scheduler/test_eval.py]: Should this have some kind of timeout? What are the chances that something could goo wrong and it will run forever?

File: tests/integration/scheduler/test_postpone.py [hunk 3]
```diff
@@ -43,10 +43,13 @@ def setUp(self):
             functions = {'test.ping': ping}
             self.schedule = salt.utils.schedule.Schedule(copy.deepcopy(DEFAULT_CONFIG), functions, returners={})
         self.schedule.opts['loop_interval'] = 1
+        self.schedule.opts['run_schedule_jobs_in_background'] = False
 
     def tearDown(self):
         self.schedule.reset()
 
+        del self.schedule
```
Comment (dwoz) [inline tests/integration/scheduler/test_postpone.py]: Why is this needed? If we need some kind of shutdown, can we call the shutdown method directly, or could we just set self.schedule to None and let it get garbage collected?

Comment (garethgreenaway) [reply tests/integration/scheduler/test_postpone.py]: The `reset` function that is being called does set the schedule to an empty dict.  I added the del because the test suit kept recommending it in a log.  I'll take it out.

File: tests/integration/scheduler/test_eval.py (hunk 2 above)
Comment (garethgreenaway) [reply tests/integration/scheduler/test_eval.py]: Since the jobs are no longer running in the background this might not be necessary.  Will check and update.

File: tests/integration/scheduler/test_postpone.py (hunk 3 above)
Comment (dwoz) [reply tests/integration/scheduler/test_postpone.py]: Oh, right. I forgot the test suite will do that. I'm fine with leaving it in that case

Comment (garethgreenaway) [reply tests/integration/scheduler/test_postpone.py]: I took it out and having it as an empty dict is fine, it will be recreated in between each test run.

File: tests/integration/scheduler/test_eval.py (hunk 2 above)
Comment (garethgreenaway) [reply tests/integration/scheduler/test_eval.py]: Looks like the previous failures on py2-centos-7 are passing with the change to not have the jobs run in the background and don't require the check to see if the job is still running.  This makes sense since it's running it right away and returning right away.

**Review by dwoz — APPROVED** [review]
